sunkit_instruments/suvi/io.py

```diff
@@ -0,0 +1,320 @@
+        filenames, e.g. "-L1b-Fe171" for a 171 L1b file and "-l2-ci171"
+        for a 171 L2 HDR composite. If those patterns are not found
+        in the filename, the files will not be recognized.
+
+    .. note::
+        If input_filename is an L1b netCDF file, the information from
+        the netCDF file is transformed into a FITS header.
+
+    Parameters
+    ----------
+    input_filename: `str`
+        File to read.
+
+    return_DQF: `bool`, optional. Default: False.
+        If True, also returns the data quality flag array (DQF, only for L1b files).
+
+    return_header_only: `bool`, optional. Default: False.
+        If True, does not return the data array (but optionally the DQF).
+
+    Returns
+    -------
+    header: `astropy.io.fits.header.Header`
+        header, if return_header_only is True.
+
+    header, data: `astropy.io.fits.header.Header`, `~numpy.ndarray`
+        header and data.
+
+    header, dqf: `astropy.io.fits.header.Header`, `~numpy.ndarray`
+        header and data quality flags, if both return_header_only and return_DQF
+        are True.
+
+    header, data, dqf: `astropy.io.fits.header.Header`, `~numpy.ndarray`, `~numpy.ndarray`
+        header, data, and data quality flags if return_DQF is True.
+    """
+    fits_file_extensions = ('.fits', '.fts', '.fits.gz', 'fts.gz', 'fits.bz', 'fts.bz')
+    netCDF_file_extensions = ('.nc', '.nc.gz', '.nc.bz', '.cdf', '.cdf.gz', '.cdf.bz')
+
+    composite_matches = ['-l2-ci094', '-l2-ci131', '-l2-ci171', '-l2-ci195', '-l2-ci284', '-l2-ci304']
+    L1b_matches = ['-L1b-Fe093', '-L1b-Fe131', '-L1b-Fe171', '-L1b-Fe195', '-L1b-Fe284', '-L1b-He303']
+
+    IS_COMPOSITE = False
+
+    # If it is a fits file, it is easy!
+    if input_filename.lower().endswith(fits_file_extensions):
+        # Based on the filename, determine if we are dealing
+        # with L1b files or HDR composites (or neither).
+        if any(fn in os.path.basename(input_filename) for fn in composite_matches):
+            if return_header_only:
+                header = fits.getheader(input_filename, 1)
+            else:
+                hdu = fits.open(input_filename)
+                data, header = hdu[1].data, hdu[1].header
+                hdu.close()
+            IS_COMPOSITE = True
+        elif any(fn in os.path.basename(input_filename) for fn in L1b_matches):
+            hdu = fits.open(input_filename)
+            if return_header_only:
+                header = fix_L1b_header(input_filename)
+            else:
+                hdu = fits.open(input_filename)
+                data, header = hdu[0].data, fix_L1b_header(input_filename)
+            if return_DQF:
+                dqf = hdu[1].data
+            hdu.close()
+        else:
+            raise ValueError("File "+input_filename+" does not look like a SUVI L1b FITS file or L2 HDR composite.")
+
+    # If it is a netCDF file on the other hand...
+    elif input_filename.lower().endswith(netCDF_file_extensions):
```

Comment:
Since the read function handles both FITS and NETCDF, I think it would be better if we take the blocks of code that handle each file format and create a `_read_netcdf`, `_read_fits` that are called within this main read function. That will allow me to review the parts of each file format separately.

Reply:
Ok, I can do that.